File: Doc/NoGraphicsExample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace NoGraphicsExample
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        #region Form
        private void CreateForm()
        {
            VectorDraw.Professional.vdFigures.vdRect rect = new VectorDraw.Professional.vdFigures.vdRect();
            rect.SetUnRegisterDocument(vdDocumentComponent1.Document);
            rect.setDocumentDefaults();
            rect.InsertionPoint = new VectorDraw.Geometry.gPoint();
            rect.Width = 120;
            rect.Height = 240;
            vdDocumentComponent1.Document.Model.Entities.AddItem(rect);

            rect = new VectorDraw.Professional.vdFigures.vdRect();
            rect.SetUnRegisterDocument(vdDocumentComponent1.Document);
            rect.setDocumentDefaults();
            rect.InsertionPoint = new VectorDraw.Geometry.gPoint(-5.0,-5.0);
            rect.Width = 130.0;
            rect.Height = 250.0;
            vdDocumentComponent1.Document.Model.Entities.AddItem(rect);

            rect = new VectorDraw.Professional.vdFigures.vdRect();
            rect.SetUnRegisterDocument(vdDocumentComponent1.Document);
            rect.setDocumentDefaults();
            rect.InsertionPoint = new VectorDraw.Geometry.gPoint(65.0, 0.0);
            rect.Width = 55.0;
            rect.Height = 30.0;
            vdDocumentComponent1.Document.Model.Entities.AddItem(rect);

            VectorDraw.Professional.vdFigures.vdMText Signature = new VectorDraw.Professional.vdFigures.vdMText();
            Signature.SetUnRegisterDocument(vdDocumentComponent1.Document);
            Signature.setDocumentDefaults();
            Signature.InsertionPoint = new VectorDraw.Geometry.gPoint(93.0, 25.0);
            Signature.TextString = "\\C1;VectorDraw\nPlatonos 1 - Dafni - 17234\nPhone : 210-9739781\nFAX:210-9739159";
            Signature.HorJustify = VectorDraw.Professional.Constants.VdConstHorJust.VdTextHorCenter;
            vdDocumentComponent1.Document.Model.Entities.AddItem(Signature);

            UpdatePicture();
        }
        private void UpdatePicture()
        {
            if (picPrev1.Width == 0 || picPrev1.Height == 0) return;

            vdDocumentComponent1.Document.CommandAction.CmdSelect("ALL");
            VectorDraw.Professional.vdCollections.vdSelection selset = vdDocumentComponent1.Document.Selections.FindName("VDRAW_PREVIOUS_SELSET");
            VectorDraw.Geometry.Box Bbox = selset.GetBoundingBox();
            Bbox.AddWidth(5.0);

            Image img1 = new Bitmap(picPrev1.Width, picPrev1.Height);
            System.Drawing.Graphics graph = Graphics.FromImage(img1);
            Bbox.TransformBy(vdDocumentComponent1.Document.ActiveLayOut.World2ViewMatrix);
            vdDocumentComponent1.Document.ActiveLayOut.RenderToGraphics(graph, Bbox, img1.Width, img1.Height);
            picPrev1.Image = img1;
        }
        #endregion

        #region Buttons
        private void butExit_Click(object sender, EventArgs e)
        {
            this.Close();
        }
        private void butExport_Click(object sender, EventArgs e)
        {
            vdDocumentComponent1.Document.SaveAs(Application.ExecutablePath.Substring(0, Application.ExecutablePath.Length - 4) + ".svg");
            vdDocumentComponent1.Document.SaveAs(Application.ExecutablePath.Substring(0, Application.ExecutablePath.Length - 4) + ".bmp");
            vdDocumentComponent1.Document.SaveAs(Application.ExecutablePath.Substring(0, Application.ExecutablePath.Length - 4) + ".jpg");
            vdDocumentComponent1.Document.SaveAs(Application.ExecutablePath.Substring(0, Application.ExecutablePath.Length - 4) + ".png");

            vdDocumentComponent1.Document.Model.Printer.PrinterName = Application.ExecutablePath.Substring(0, Application.ExecutablePath.Length - 4) + ".pdf";
            vdDocumentComponent1.Document.Model.Printer.PrintExtents();
            vdDocumentComponent1.Document.Model.Printer.PrintWindow.AddWidth(5.0);
            vdDocumentComponent1.Document.Model.Printer.PrintScaleToFit();
            vdDocumentComponent1.Document.Model.Printer.PrintOut();

            vdDocumentComponent1.Document.Model.Printer.PrinterName = Application.ExecutablePath.Substring(0, Application.ExecutablePath.Length - 4) + ".emf";
            vdDocumentComponent1.Document.Model.Printer.PrintExtents();
            vdDocumentComponent1.Document.Model.Printer.PrintScaleToFit();
            vdDocumentComponent1.Document.Model.Printer.PrintOut();

            MessageBox.Show("6 Files have been created : PDF,EMF,SVG,BMP,JPG,PNG to tha application's path.");
        }
        #endregion

        #region Events
        private void Form1_Load(object sender, EventArgs e)
        {
            CreateForm();
            MessageBox.Show("This sample does not contain any .NET graphical component. It contains only a vdDocument component which does all the work.");
        }

        private void Form1_Resize(object sender, EventArgs e)
        {
            UpdatePicture();
        }
        #endregion
    }
}

File: VB/Doc/NoGraphicsExample_VB/Form1.vb
Public Class Form1
    Private Sub ButExit_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ButExit.Click
        Application.Exit()
    End Sub

    Private Sub ButExport_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ButExport.Click
        VdDocumentComponent1.Document.SaveAs(Application.ExecutablePath.Substring(0, Application.ExecutablePath.Length - 4) + ".svg")
        VdDocumentComponent1.Document.SaveAs(Application.ExecutablePath.Substring(0, Application.ExecutablePath.Length - 4) + ".bmp")
        VdDocumentComponent1.Document.SaveAs(Application.ExecutablePath.Substring(0, Application.ExecutablePath.Length - 4) + ".jpg")
        VdDocumentComponent1.Document.SaveAs(Application.ExecutablePath.Substring(0, Application.ExecutablePath.Length - 4) + ".png")

        VdDocumentComponent1.Document.Model.Printer.PrinterName = Application.ExecutablePath.Substring(0, Application.ExecutablePath.Length - 4) + ".pdf"
        VdDocumentComponent1.Document.Model.Printer.PrintExtents()
        VdDocumentComponent1.Document.Model.Printer.PrintWindow.AddWidth(5.0)
        VdDocumentComponent1.Document.Model.Printer.PrintScaleToFit()
        VdDocumentComponent1.Document.Model.Printer.PrintOut()

        VdDocumentComponent1.Document.Model.Printer.PrinterName = Application.ExecutablePath.Substring(0, Application.ExecutablePath.Length - 4) + ".emf"
        VdDocumentComponent1.Document.Model.Printer.PrintExtents()
        VdDocumentComponent1.Document.Model.Printer.PrintScaleToFit()
        VdDocumentComponent1.Document.Model.Printer.PrintOut()

        MessageBox.Show("6 Files have been created : PDF,EMF,SVG,BMP,JPG,PNG to tha application's path.")
    End Sub

    Private Sub CreateForm()
        Dim rect As VectorDraw.Professional.vdFigures.vdRect
        rect = New VectorDraw.Professional.vdFigures.vdRect()
        rect.SetUnRegisterDocument(VdDocumentComponent1.Document)
        rect.setDocumentDefaults()
        rect.InsertionPoint = New VectorDraw.Geometry.gPoint()
        rect.Width = 120
        rect.Height = 240
        VdDocumentComponent1.Document.Model.Entities.AddItem(rect)

        rect = New VectorDraw.Professional.vdFigures.vdRect()
        rect.SetUnRegisterDocument(VdDocumentComponent1.Document)
        rect.setDocumentDefaults()
        rect.InsertionPoint = New VectorDraw.Geometry.gPoint(-5.0, -5.0)
        rect.Width = 130.0
        rect.Height = 250.0
        VdDocumentComponent1.Document.Model.Entities.AddItem(rect)

        rect = New VectorDraw.Professional.vdFigures.vdRect()
        rect.SetUnRegisterDocument(VdDocumentComponent1.Document)
        rect.setDocumentDefaults()
        rect.InsertionPoint = New VectorDraw.Geometry.gPoint(65.0, 0.0)
        rect.Width = 55.0
        rect.Height = 30.0
        VdDocumentComponent1.Document.Model.Entities.AddItem(rect)

        Dim Signature As VectorDraw.Professional.vdFigures.vdMText
        Signature = New VectorDraw.Professional.vdFigures.vdMText()
        Signature.SetUnRegisterDocument(VdDocumentComponent1.Document)
        Signature.setDocumentDefaults()
        Signature.InsertionPoint = New VectorDraw.Geometry.gPoint(93.0, 25.0)
        Signature.TextString = "\C1;VectorDraw\NPlatonos 1 - Dafni - 17234\NPhone : 210-9739781\NFAX:210-9739159"
        Signature.HorJustify = VectorDraw.Professional.Constants.VdConstHorJust.VdTextHorCenter
        VdDocumentComponent1.Document.Model.Entities.AddItem(Signature)

        UpdatePicture()
    End Sub
    Private Sub UpdatePicture()

        If (picPrev1.Width = 0 Or picPrev1.Height = 0) Then Exit Sub

        VdDocumentComponent1.Document.CommandAction.CmdSelect("ALL")
        Dim selset As VectorDraw.Professional.vdCollections.vdSelection
        selset = VdDocumentComponent1.Document.Selections.FindName("VDRAW_PREVIOUS_SELSET")
        Dim Bbox As VectorDraw.Geometry.Box
        Bbox = selset.GetBoundingBox()
        Bbox.AddWidth(5.0)

        Dim img1 As Image
        img1 = New Bitmap(picPrev1.Width, picPrev1.Height)
        Dim graph As System.Drawing.Graphics
        graph = Graphics.FromImage(img1)
        Bbox.TransformBy(VdDocumentComponent1.Document.ActiveLayOut.World2ViewMatrix)
        VdDocumentComponent1.Document.ActiveLayOut.RenderToGraphics(graph, Bbox, img1.Width, img1.Height)
        picPrev1.Image = img1
    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        CreateForm()
        MessageBox.Show("This sample does not contain any .NET graphical component. It contains only a vdDocument component which does all the work.")
    End Sub

    Private Sub Form1_Resize(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Resize
        UpdatePicture()
    End Sub
End Class
